src/cloud_provider/aws/kubernetes/eks.rs

```diff
@@ -669,3 +707,48 @@ impl Listen for EKS {
         self.listeners.push(listener);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::cloud_provider::aws::kubernetes::eks::EKS;
+    use crate::cloud_provider::models::NodeGroups;
+    use crate::errors::Tag;
+    use crate::events::{EventDetails, InfrastructureStep, Stage, Transmitter};
+    use crate::io_models::QoveryIdentifier;
+
+    #[test]
+    fn test_allowed_eks_nodes() {
```

Comment:
OPT: Maybe add more test cases:
- test all allowed instances types
- test all not allowed instances types

It's ok to test only one unsupported.

Also maybe if you test all, maybe split into 3 tests:
-  `test_allowed_eks_nodes`
- `test_unallowed_eks_nodes`
- `test_unsupported_eks_nodes`